src/bam/collapse_reads_to_fragments/calc_consensus.rs

```diff
@@ -79,6 +87,41 @@ impl<'a> CalcOverlappingConsensus<'a> {
     fn overlap(&self) -> usize {
         self.overlap
     }
+    fn build_consensus_strand(
+        &self,
+        consensus_strand: &mut Vec<u8>,
+        ref_base: u8,
+        base_pos: usize,
+    ) {
+        let mut strands = HashSet::new();
```

Comment:
I'd rather have an enum for this:

```
use std::ops::BitOrAssign;

#[derive(Eq, PartialEq)]
enum StrandObservation {
    None,
    Forward,
    Reverse,
    Both,
}

impl BitOrAssign for StrandObservation {
    fn bitor_assign(&mut self, rhs: Self) {
        if let StrandObservation::None = self {
            *self = rhs;
        } else if *self != rhs {
            *self = StrandObservation::Both;
        }
    }
}
```

Reply:
Then, you can just initialize with `let mut strand = StrandObservation::None`, and then update with `strand |= obs`.